src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int INDEX_OF_NAME_IN_DATA = 1;
+    private static final int INDEX_OF_WORK_HOURS_IN_DATA = 2;
+    private static final int INDEX_OF_SALARY_FOR_DAY_IN_DATA = 3;
+    private static final String REGEX = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            int sum = 0;
+            for (int j = 0; j < data.length; j++) {
+                if (name.equals(data[j].split(REGEX)[INDEX_OF_NAME_IN_DATA])) {
+                    String[] splitData = data[j].split(REGEX);
+
+                    LocalDate convertDate = LocalDate.parse(splitData[0], FORMATTER);
+                    LocalDate fromDate = LocalDate.parse(dateFrom, FORMATTER);
+                    LocalDate toDate = LocalDate.parse(dateTo, FORMATTER);
```

Comment:
these variables could be moved before for loops because it has the same value for each loop iteration